assets/src/web-stories-block/block/transforms.js

```diff
@@ -133,6 +137,47 @@ const transforms = {
         },
       },
     },
+    {
+      type: 'block',
+      blocks: ['core/legacy-widget'],
+      isMatch: ({ idBase, instance }) => {
+        if (!instance?.raw) {
+          // Can't transform if raw instance is not shown in REST API.
+          return false;
+        }
+        return idBase === 'web_stories_widget';
+      },
+      transform: ({ instance }) => {
+        const transformedBlock = createBlock('web-stories/embed', {
+          blockType: 'latest-stories',
+          viewType: instance.raw.view_type,
+          fieldState: {
+            show_title: instance.raw.show_title,
+            show_author: instance.raw.show_author,
+            show_date: instance.raw.show_date,
+            show_excerpt: instance.raw.show_excerpt,
+            show_archive_link: instance.raw.show_archive_link,
+            show_sharp_corners: instance.raw.sharp_corners,
+            show_image_alignment: instance.raw.image_alignment,
+          },
+          archiveLinkLabel: instance.raw.archive_link_label,
+          circleSize: instance.raw.circle_size,
+          numOfColumns: instance.raw.number_of_columns,
+          numOfStories: instance.raw.number_of_stories,
+          orderby: instance.raw.orderby.replace('post_', ''),
+          order: instance.raw.order.toLowerCase(),
+        });
+        if (!instance.raw?.title) {
+          return transformedBlock;
+        }
+        return [
+          createBlock('core/heading', {
+            content: instance.raw.title,
+          }),
+          transformedBlock,
+        ];
```

Comment:
Example for https://github.com/WordPress/gutenberg/blob/152420f8d3b49163d46be9e42111dbb1893eab3f/packages/block-library/src/legacy-widget/transforms.js#L163-L186